### PR TITLE
Generate a library with single SVGs and Components based on the Icomoon library

### PR DESCRIPTION
**Pull request by mt-roberto**
## Description

On `postinstall` it generates single SVGs and single React Components with their Typescript definition from the Icomoon library.

### Icon component with icon name:

```javascript
import React from 'react';
import { ReactElement } from 'react';
const Icon = 'mt-web-icons/components/react/Icon';

export default function IconAuto(): ReactElement {
  return <Icon name="auto" width={100} height={100} fill="#ffffff" />;
}
```

### Icon component with invalid icon name will display a placeholder icon:

```javascript
import React from 'react';
import { ReactElement } from 'react';
const Icon = 'mt-web-icons/components/react/Icon';

export default function IconAuto(): ReactElement {
  return <Icon name="Invalid Icon" />;
}
```

### Icon with defined placeholder and fallback for suspence:

```javascript
import React from 'react';
import { ReactElement } from 'react';
const Icon = 'mt-web-icons/components/react/Icon';

export default function IconAuto(): ReactElement {
  return <Icon name="Auto" placeholder={null} fallback={<div />} />;
}
```

### Specific Icon:

```javascript
import React from 'react';
import { ReactElement } from 'react';
import IconAuto from 'mt-web-icons/components/react/icons/Auto';

export default function Icon(): ReactElement {
  return <IconAuto width={100} height={100} fill="#ffffff" />;
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,45 @@
...
 7. Copy the whole unzipped folder to update this repository
+
+# How to use it
+
+```javascript
+import React from 'react';
+import { ReactElement } from 'react';
+
+export default function IconAuto(): ReactElement {
+  const Icon = lazy(() => import(`mt-web-icons/components/react/Auto`));
```
Comment (mt-max) [inline README.md]: I think it is better to move it outside of `IconAuto`, there is no need to reassign to icon all the time

File: templates/IconComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+}
+
+export default function Icon(props) {
+  const {
+    name,
+    width = 24,
+    height = 24,
+    fallback = null,
+    placeholder= <div style={{ borderRadius: 100, backgroundColor: '#000', width: `${width}px`, height: `${height}px` }} />
```
Comment (mt-max) [inline templates/IconComponent.jsx]: I am not sure if the `width` and `height` is defined at this point in the default `placeholder` value

**Comment by mt-micky** [comment]
Did you ask feedback about this implementation before developing, merging and releasing?

**Comment by mt-roberto** [comment]
Is this a rhetorical question?

**Comment by mt-micky** [comment]
Yes.
Why a post-install hook?
Why not including the generated file in the repo?
You could have taken example from the other libraries we already have.

**Pull request merged**